deployment/Dockerfile.runner-jvm.test

```diff
@@ -0,0 +1,22 @@
+# Huge hack, we want the JVM and same runtime environment as the runner
```

Comment:
We need the JVM to run the java module bytecode though right?
(This hack is from before this PR btw, it's just been moved out of the normal runner to the jvm runner)

Reply:
I don't know what the intermediate state is, but the end state is that the base runner image knows nothing about the JVM, Python, or any runtime, and each runtime-specific image layers on top of the runner image.

Re. this specific comment, I don't quite understand why the base runner would still contain the JVM, but maybe it's an intermediate step?

Reply:
Ahhh yes yes. It's pulling from `ftl0/ftl-runner` which as of this PR contains the JVM, and `ftl0/ftl-runner-jvm` has not yet been published.
Once this PR makes it to release I'll move it to pulling from the jvm runner.

Reply:
This was a hack to make the image build faster and to prevent the GHA runners running out of disk space. 

> I don't know what the intermediate state is, but the end state is that the base runner image knows nothing about the JVM, Python, or any runtime, and each runtime-specific image layers on top of the runner image.

@matt2e we should probably change the builds to use FROM <base-runner-image> instead of basically building a new base image each time. This will reduce that amount of network traffic pulling images as they will share a base layer.